File: src/lib/api/stundenplan42/class/Lesson.ts
export default class Lesson {
	id: string | undefined;
	subject: string | undefined;
	teacher: string | undefined;
}

/**
 * Parses a lesson from the JSON response
 * @param json JSON response
 */
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function fromJson(json: any): Lesson {
	const lesson = new Lesson();
	lesson.id = json.UeNr['_'];
	lesson.subject = json.UeNr.$.UeGr || json.UeNr.$.UeFa;
	lesson.teacher = json.UeNr.$.UeLe;
	return lesson;
}

File: lib/api/stundenplan24/models/lesson.dart
class Lesson {
  final String id;
  final String subject;
  final String teacher;

  Lesson(this.id, this.subject, this.teacher);

  Lesson.parseXMLJson(dynamic json)
      : id = json['UeNr']['value'],
        subject = json['UeNr']['_UeGr'] ?? json['UeNr']['_UeFa'],
        teacher = json['UeNr']['_UeLe'];

  Lesson.fromJson(dynamic json)
      : id = json['id'],
        subject = json['subject'],
        teacher = json['teacher'];

  Map<String, dynamic> toJson() => {'id': id, 'subject': subject, 'teacher': teacher};
}
